### ISSUE TITLE
Unable to Run +trouble after OTA

### ISSUE THREAD

**Issue by sipwex-barfus** [urbit/urbit#3069]
**Describe the bug**

When I type `+trouble` in the dojo I get the following returned:

```
> +trouble|
bail: 4
! /cs/~litpub/kids/1/late/foo
[%dojo-lame %arvo-response]
> +trouble
```

**To Reproduce**
I had recently(yesterday) breached starting anew with 0.10.7 due to consistent issues with my original ship. I had therefore started as "fresh" as one could prior to the OTA (797 fragments) today, having synced first to ~marzod, got no response or change, so I unsynced from ~marzod and synced to ~binzod (since my own star didn't seem to be giving me the OTA). Below are the logs from when I sync'd to ~binzod and updated (likely superfluous information but including just in case it changes anything)

```
> |sync %base ~binzod %kids
>=
activated sync from %kids on ~binzod to %base
beginning sync from %kids on ~binzod to %base 
```

wherein my ship downloaded 102,848 fragments updating from `[%base-hash 0vu.niou0.tga8g.m8jo3.pbpr0.f3ih0.vkkhu.a0i7o.dhntn.82tm9.81dn2]` to the one prior to today's.

Today a sync began for OTA with 797 fragments which appears to have some errors, logs here https://pastebin.com/ybbqdu1e 

**Expected behaviour**
+trouble should return my base hash information

**System (please supply the following information, if relevant):**
 - OS: macOS
 - Vere and Urbit OS versions: Not sure how to get this for Vere but 0.10.7 for Urbit. 
 - Your ship's `%base` hash (use `.^(@uv %cz /=base=)` to check):
```> .^(@uv %cz /=base=)
0va.2f45l.3frvo.37bn6.8u19e.cu8e5.dedks.l51sj.hljo0.6qcml.t74k1
```

**Comment by mikegogulski** [comment]
+1

**Comment by philipcmonk** [comment]
If you were never able to get updates from ~litpub, `+trouble` will fail like that.  We need to switch that from looking for parent to looking for OTA.  Is everything else about it working alright?

**Comment by Faolain** [comment]
Interesting thanks @philipcmonk , and more or less, seem to have trouble joining any chats on Urbit Community even though I'm in the group (~bitbet-bolbel). I get a spinner on restoring past messages even when left for an entire day on one channel along with many `arvo: replacement event (for %hear) failed` messages in the dojo.

**Comment by philipcmonk** [comment]
@Faolain Can you run `|verb` until you get that message again and send the stack trace?  Run `|verb` again to turn it off.

**Comment by Faolain** [comment]
Done @philipcmonk , https://pastebin.com/ZcEgyUiD should have two events in there.

**Comment by sipwex-barfus** [comment]
Interesting thanks @philipcmonk , and more or less, seem to have trouble joining any chats on Urbit Community even though I'm in the group (~bitbet-bolbel). I get a spinner on restoring past messages even when left for an entire day on one channel along with many arvo: replacement event (for %hear) failed messages in the dojo.


Update Logs: 
Should have two arvo replacement events in here https://pastebin.com/ZcEgyUiD

**Comment by philipcmonk** [comment]
You have a couple of "dangling bones" with ~litpub, which means on those flows you can't make progress.  This is a known bug (#2359), but we haven't been able to track down what causes it, our best guess is something incorrect about breach handling, I know we found one place where that caused a problem.  Our only workarounds are to breach or switch to download updates from another ship.

cc @belisarius222

**Comment by sipwex-barfus** [comment]
Ah hmm appreciate it philip. I had just breached prior to this so this was almost as fresh of an "install" as possible beyond having run `|sync %base ~binzod %kids` to get my updates from ~binzod (since ~litpub wasn't doing so for some reason). Any suggestions on how to proceed so that the same thing doesn't happen again?

**Comment by philipcmonk** [comment]
I'll take a look on ~binzod at those flows and see if I can track down what might be the cause

**Comment by philipcmonk** [comment]
Do you have the debug dashboard?  Can you run `|start %dbug`, navigate to `/~debug`, go to ames, search for ~binzod, and see if you have anything around bones 88 or 176?

When you breached, did you do anything to load old state?

**Comment by philipcmonk** [comment]
Oh wait I'm not thinking straight, ~binzod is communicating fine, it's ~litpub that we need to look up.  I forget who ~litpub is (@litpub ?), it would be great if they can do the same thing I said in the previous comment except searching for ~sipwex-barfus

**Comment by sipwex-barfus** [comment]
Not sure if this helps but it looks like I'm picking up the next OTA right now automatically (10.246 fragments downloading). Assuming this won't fix things?

When I breached didn't do anything special that I'm aware of, deleted old pier and "started fresh". 

Should I still run the above debug but searching ~litpub?

**Comment by philipcmonk** [comment]
Good that you're getting the OTA correct, I expect you'll continue to get it correctly from ~binzod.  The dangling bones aren't a huge issue for you day-to-day, but they've been a nagging problem for months and now that we have the debug dashboard we can localize the problem.  I would still like to see if you have anything near those bones, yes, but ~litpub's side will have more info.  I reached out on ~litpub/ops.

Basically ~litpub is sending messages on a flow (bone) you don't know anything about, which suggests that it didn't completely forget everything about you when you breached, which is what it's supposed to do.  Seeing the flow from ~litpub's perspective should tell us what didn't forget properly.

**Comment by sipwex-barfus** [comment]
Appreciate it @philipcmonk, not necessarily sure what I should be looking for here but will paste the results. 

![Screen Shot 2020-06-30 at 5 47 15 PM](https://user-images.githubusercontent.com/62186759/86180289-d0261080-baf9-11ea-898d-a297af93737f.png)

**Comment by litpub** [comment]
You can ping with @litpub - I'm monitoring ~ops more actively now too, saw your message there.

> Can you run |start %dbug, navigate to /~debug, go to ames, search for ~sipwex-barfus, and see if you have anything around bones 88 or 176?

I flat out do not see anything for bones 88 or 176.

**Comment by philipcmonk** [comment]
Bones are flipped around some, i guess these are backward bones (from ~litpub's perspective), so 89 and 177 would be the relevant ones.  Can you expand the `snd` and `rcv` tabs for 89 and 177?

At any rate, ~litpub seems to have a lot more bones for ~sipwex than the reverse, which is very strange.  ~litpub, can you run `+keys ~sipwex-barfus`?  The other thing I'm wondering is whether last time ~sipwex breached it was in fact a breach and not just cycling the keys.

**Comment by litpub** [comment]
snd/rcv 89:

<img width="623" alt="Screen Shot 2020-06-30 at 7 42 00 PM" src="https://user-images.githubusercontent.com/61065763/86187450-0d3dd300-baf9-11ea-8b90-9e5bb7d484ef.png">

snd/rcv 177:

<img width="610" alt="Screen Shot 2020-06-30 at 7 42 13 PM" src="https://user-images.githubusercontent.com/61065763/86187451-0f079680-baf9-11ea-93d1-902f74852b9a.png">

```
> +keys ~sipwex-barfus
[life=[~ 4] rift=[~ 2]]
```

which seems to match what's on chain...

**Comment by philipcmonk** [comment]
https://etherscan.io/tx/0x0709c36fc7f89635137a6de179986f3ae5165a7ca44df9fac93fd0ae193da758

It looks to me like the last time time ~sipwex-barfus changed keys they did *not* breach (note `discontinuous` is 0), which would explain why people they had talked are still trying to send the same messages to them.  This happens occasionally, so possibly this should be called out in Bridge more loudly.

@sipwex-barfus You will need to breach again, and make certain sure the checkbox for "breaching" or "discontinuous" is checked.  Without that it changes your keys without actually breaching.

**Comment by sipwex-barfus** [comment]
Thanks so much everyone, so sorry for the wild goose chase. I guess with so many things going on/OTA's going out, breaches etc it became somewhat easy to conflate what could be causing what. Really appreciate it again.

As for the instructions it could be possible I didn't appropriately breach, I have a suggestion below for the instructions since they seem to differ from what is actually displayed on bridge. It appears there are two routes with similar names "Reset Keys" within ID and "Reset Networking Keys" within OS. I may have likely gone with the first (hence no breach). Thanks again!  

Current Instructions on https://urbit.org/docs/tutorials/guide-to-breaches/
- Go to bridge.urbit.org and log into your identity.
- Go to the Admin section. Then go to Edit Permissions, then Networking Keys.
- Check the Breach Continuity box. Click Reset Networking Keys, and then click Send Transaction.
- Click Download Arvo Keyfile.
- Delete or archive your old pier.
- Create a new pier by booting your ship with your new keyfile.
- Rejoin your favorite chat channels and subscriptions.

Suggested Instructions
- Go to bridge.urbit.org and log into your identity.
- Scroll to the bottom of the page and select `OS (Urbit OS Settings)`. Then click `Reset Networking` Keys, and check the Breach Continuity box.
- Click `Reset Networking Keys`, and then click `Send Transaction`, if you see an error of not enough ETH, please send only the indicated amount of ether to the address presented.
- Click Download Arvo Keyfile.
- Delete or archive your old pier.
- Create a new pier by booting your ship with your new keyfile.
- Rejoin your favorite chat channels and subscriptions.

**Comment by philipcmonk** [comment]
Those look good to me, I might even underline "check the Breach Continuity box" or make it its own bullet.

**Comment by sipwex-barfus** [comment]
Went to edit the docs but they appear to be updated already in the repo, https://urbit.org/docs/tutorials/guide-to-breaches/ vs https://github.com/urbit/docs/blob/bc78fd93090eaf3699568e0267ef8d0f4145d7b2/tutorials/guide-to-breaches.md . Looks like someone on the urbit team just needs to push those changes to the live site (CI in the future ftw so that it's automatic :) )

**Comment by sipwex-barfus** [comment]
So sorry to chime in here again, I let the OTA run (110k fragments) so I stepped away and now seem to get the same error more or less. I'm positive I selected the checkbox to breach this time on a totally fresh pier (eth tx has discontinuous as true). Now I still get many `arvo: replacement event (for %hear) failed`. Looking through the scrollback on the terminal I see the following after the first 110k fragment update:

```
kiln: OTA to %kids failed from %kids on ~litpub
%merge-no-merge-base
consider a %this or %that merge to get a mergebase
ames: ~litpub: send dupe message ack 2 ok=%.y
ames: ~litpub: send dupe message ack 1 ok=%.y
kiln: downloading OTA update from %kids on ~litpub
```

It then looks like another 10.315 fragment OTA downloaded but also got this error after finishing:

```
ames: ~litpub: send dupe message ack 1 ok=%.y
kiln: finished downloading OTA from %kids on ~litpub
kiln: applying OTA to %home from %kids on ~litpub
kiln: OTA to %home succeeded from %kids on ~litpub
kiln: applying OTA to %kids from %kids on ~litpub
```

Running +trouble, +keys fails, I turned my urbit off and back on and now +trouble works. 

+trouble
```
[ [%base-hash 0vs.9288r.bdja5.ic2br.jigbg.0kduq.stbiu.npcsj.mtsk0.dhf80.l4gq1]
  [%home-hash 0vs.9288r.bdja5.ic2br.jigbg.0kduq.stbiu.npcsj.mtsk0.dhf80.l4gq1]
  [%kids-hash 0v1m.ol9aa.mpns9.d88d8.pb4td.v4mjb.umdps.nui8r.cscp4.qrou6.cms4k]
  [%our ship=~sipwex-barfus point='2865038758' life=[~ 6] rift=[~ 4]]
  [%sponsor ship=~litpub point='1446' life=[~ 3] rift=[~ 0]]
  [%dopzod ship=~dopzod point='4608' life=[~ 3] rift=[~ 2]]
  "Compare lifes and rifts to values here:"
  "https://etherscan.io/address/azimuth.eth#readContract"
  "  life - getKeyRevisionNumber"
  "  rift - getContinuityNumber"
  ~
]
```

+keys
```
>   dojo: nest-need
{{now/@da @ our/@p * *} {her/@p $~} $~}
>   dojo: nest-have
{{now/@da eny/@uvJ bec/{p/@p q/@tas r/?({$da p/@da} {$tas p/@tas} {$ud p/@ud})}} $~ $~}
> +keys
dojo: generator failure
```

running |verb to catch the arvo replacement failed issue have a few in this logfile
https://pastebin.com/qitMpau9

**Comment by belisarius222** [comment]
Can you run this?

=f -build-file /lib/sole/hoon

If you can, then you've received the ford fusion OTA and your system is at least somewhat coherent. If not, we'll have to look into it in more detail.

—
~rovnys-ricfer
https://urbit.org

**Comment by sipwex-barfus** [comment]
Just ran it @belisarius222 , no output(no error, nothing just goes back to terminal). Is this a good sign?

**Comment by belisarius222** [comment]
Yes, it is! Sounds like you've entered the new world after all.

—
~rovnys-ricfer
https://urbit.org

**Comment by sipwex-barfus** [comment]
Thanks again! So happy to hear :) very excited

**Comment by mikegogulski** [comment]
I was having the same problem as ~sipwex-barfus. +trouble didn't work, and didn't receive the OTA for v1.0.25.

I tried the breach/rekey/relaunch instructions as above, and now I've got a different problem on booting:
```
~/urbit-linux64-v0.10.3$ ./urbit -w nocdys -k ../nocdys-4.key  
~
urbit 0.10.3
boot: home is /home/syadasti/urbit-linux64-v0.10.3/nocdys
loom: mapped 2048MB
lite: arvo formula 69dfe787
lite: core 31c5cfd7
lite: final state 31c5cfd7
boot: downloading pill https://bootstrap.urbit.org/urbit-v0.10.3.pill
boot: retrieving latest block
error fetching https://mainnet.infura.io/v3/196a7f37c7d54211b4a07904ec73ad87: HTTP 401
```

**Comment by Fang-** [comment]
@mikegogulski you're going to want to install the latest binary, and boot with that. We're on 0.10.7 now. It has a bunch of improvements, and also fixes the issue you're running into right now.

**Comment by mikegogulski** [comment]
> @mikegogulski you're going to want to install the latest binary, and boot with that. We're on 0.10.7 now. It has a bunch of improvements, and also fixes the issue you're running into right now.

Thanks, that fixed it.

**Comment by rbpaul** [comment]
I am also getting this "error" running trouble:

~motlyd-lonmes-walmer-bolfet:dojo> +trouble|
bail: 4
! /cs/~walmer-bolfet/kids/1/late/foo
[%dojo-lame %arvo-response]
! /cs/~walmer-bolfet/kids/1/late/foo
[%dojo-lame %arvo-response]
! /cs/~walmer-bolfet/kids/1/late/foo
[%dojo-lame %arvo-response]
> +trouble

Any fix?

**Comment by philipcmonk** [comment]
This is fixed in an OTA, but not in the pill, so it'll still crash until you get your first OTA.  We'll probably push a new pill later this week.

**Comment by Beasta** [comment]
I'm having this issue. 
When I `+trouble` I get
```
+trouble
bail: 4

bail: 2
[%dojo-lame %arvo-response]
```
I've tried switching from my sponsor (~litdev) to ~morzod for system updates and it doesn't fix it. 

When I type `+vats` i get 
```
+vats
bail: 2
[%dojo-lame %arvo-response]
> +vats
```
When I type `+vat %base` i get 
```
